db/db_basic_test.cc

```diff
@@ -3339,7 +3342,7 @@ TEST_P(DBBasicTestDeadline, IteratorDeadline) {
       Iterator* iter = dbfull()->NewIterator(ro);
       int count = 0;
       iter->Seek("k50");
-      while (iter->Valid() && count++ < 100) {
+      while (iter->Valid() && iter->status().ok() && count++ < 100) {
```

Comment:
The iterator interface now requires !Valid() if status is not OK: https://github.com/facebook/rocksdb/wiki/Iterator#error-handling